integrationtests/controller/clustergroup/status_test.go

```diff
@@ -0,0 +1,68 @@
+package clustergroup
+
+import (
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	"github.com/rancher/fleet/integrationtests/utils"
+	"github.com/rancher/fleet/pkg/apis/fleet.cattle.io/v1alpha1"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+var _ = Describe("ClusterGroup Status Fields", func() {
+
+	var (
+		clusterName = "test-cluster"
+		groupName   = "test-cluster-group"
+	)
+
+	BeforeEach(func() {
+		var err error
+		namespace, err = utils.NewNamespaceName()
+		Expect(err).ToNot(HaveOccurred())
+
+		ns := &corev1.Namespace{ObjectMeta: metav1.ObjectMeta{Name: namespace}}
+		Expect(k8sClient.Create(ctx, ns)).ToNot(HaveOccurred())
+
+		DeferCleanup(func() {
+			Expect(k8sClient.Delete(ctx, ns)).ToNot(HaveOccurred())
+		})
+	})
+
+	When("Cluster is added", func() {
+		BeforeEach(func() {
+			clusterGroup, err := createClusterGroup(groupName, namespace, &metav1.LabelSelector{
+				MatchLabels: map[string]string{},
+			})
+			Expect(err).NotTo(HaveOccurred())
+			Expect(clusterGroup).To(Not(BeNil()))
+		})
+
+		It("updates the status fields", func() {
+			clusterGroup := &v1alpha1.ClusterGroup{}
+			err := k8sClient.Get(ctx, types.NamespacedName{Namespace: namespace, Name: groupName}, clusterGroup)
+			Expect(err).NotTo(HaveOccurred())
+			Expect(clusterGroup).To(Not(BeNil()))
+			Expect(clusterGroup.Status.ClusterCount).To(Equal(0))
+			Expect(clusterGroup.Spec.Selector.MatchLabels).To(BeNil())
+
+			cluster, err := createCluster(clusterName, namespace, nil, namespace)
+			Expect(err).NotTo(HaveOccurred())
+			Expect(cluster).To(Not(BeNil()))
+
+			Eventually(func() error {
+				err := k8sClient.Get(ctx, types.NamespacedName{Namespace: namespace, Name: groupName}, clusterGroup)
+				Expect(err).NotTo(HaveOccurred())
+				clusterGroup.Spec.Selector.MatchLabels = map[string]string{
+					"Name": clusterName,
+				}
+				return k8sClient.Update(ctx, clusterGroup)
+			}).ShouldNot(HaveOccurred())
+			Expect(clusterGroup.Status.ClusterCount).To(Equal(1))
```

Comment:
Same observation than for other cases:
I think we could check for more Status fields here, not just `ClusterCount` and `Spec.Selector.MatchLabels`

Reply:
I've added some other cases to check `status.display` as we decided to keep the scope of this PR on summary and display